code/datums/status_effects/neutral.dm

```diff
@@ -135,3 +135,159 @@
 	name = "Holding Up"
 	desc = "You're currently pointing a gun at someone."
 	icon_state = "aimed"
+
+// this status effect is used to negotiate the high-fiving capabilities of all concerned parties
+/datum/status_effect/high_fiving
+	id = "high_fiving"
+	duration = -1
+	tick_interval = -1
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = null
+	/// The carbons who were offered the ability to partake in the high-five
+	var/list/possible_takers
+	/// The actual slapper item
+	var/obj/item/slapper/slap_item
+
+/datum/status_effect/high_fiving/on_creation(mob/living/new_owner, obj/item/slap)
+	. = ..()
+	if(!.)
+		return
+
+	slap_item = slap
+	owner.visible_message("<span class='notice'>[owner] raises [owner.p_their()] arm, looking for a high-five!</span>", \
+		"<span class='notice'>You post up, looking for a high-five!</span>", null, 2)
+
+	for(var/i in orange(1, owner))
+		var/mob/living/carbon/possible_taker = i
```

Comment:
```suggestion
	for(var/mob/living/carbon/possible_taker in orange(1, owner))
```
This is one of those cases in where you want to type-check during the loop, else this will return all sorts of atoms.
Plus, byond does some optimizations for istype checks on for loops, supposedly.